src/Symfony/Component/Form/Form.php

```diff
@@ -206,19 +231,19 @@ public function getConfig()
      */
     public function getName()
     {
-        return $this->config->getName();
+        return $this->name;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getPropertyPath()
     {
-        if (null !== $this->config->getPropertyPath()) {
-            return $this->config->getPropertyPath();
+        if ($this->propertyPath || $this->propertyPath = $this->config->getPropertyPath()) {
+            return $this->propertyPath;
         }
 
-        if (null === $this->getName() || '' === $this->getName()) {
+        if (!$this->name) {
```

Comment:
what if the name is `'0'`? :smile: Can it happen?

Reply:
Yes it can, nice catch!

Reply:
Shall we add a testcase or is there already one?

Reply:
Ok so I would just write `if (trim($this->name) === '')`

Reply:
No function call :)